pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -755,6 +770,21 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 	}
 }
 
+// Return trace sampling if set correctly, or default if not.
+func getTraceSampling() float64 {
+	if traceSampling == "" {
+		return traceSamplingDefault
+	}
+	f, err := strconv.ParseFloat(traceSampling, 64)
+	if err != nil {
+		return traceSamplingDefault
+	}
+	if f < 0.0 || f > 100.0 {
+		return traceSamplingDefault
```

Comment:
should this result in a log somewhere?

Reply:
I'd be careful to log here, since the outer function is called numerous times. A metric might be useful though.

Reply:
Oops, just put in, maybe a check on startup?